pkg/controller/persistentvolume/persistentvolume_recycler_controller.go

```diff
@@ -96,16 +121,57 @@ func NewPersistentVolumeRecycler(kubeClient client.Interface, syncPeriod time.Du
 	return recycler, nil
 }
 
+func (recycler *PersistentVolumeRecycler) shouldRecycle(pv *api.PersistentVolume) error {
+	if pv.Spec.ClaimRef == nil {
+		return fmt.Errorf("Volume does not have a reference to claim")
+	}
+	if pv.Status.Phase == api.VolumeReleased {
+		// This is the first attempt to recycle, allow.
+		return nil
```

Comment:
I rewrote the patch to keep the volume `Released` and mark it as `Failed` only when all attempts to recycle/delete volume fail.